### PR TITLE
CDL Pagination

### PR DESCRIPTION
**Pull request by mayyagoldman**
https://jira-dc.paloaltonetworks.com/browse/CIAC-10072

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.yml [hunk 1]
```diff
@@ -1992,7 +2088,7 @@ script:
       type: String
   - description: Use this command in case your authentication calls fail due to internal call-limit, the command will reset the limit cache.
     name: cdl-reset-authentication-timeout
-  dockerimage: demisto/python_pancloud_v2:1.0.0.100429
+  dockerimage: devdemisto/python_pancloud_v2:1.0.0.105803 #TODO FILL IN ONCE MERGING THE NEW DOCKER IMAGE
```
Comment (ShirleyDenkberg) [inline Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.yml:2091]: Don't forget to change this!

File: Packs/CortexDataLake/ReleaseNotes/1_4_12.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+#### Integrations
+
+##### Strata Logging Service XSOAR Connector
+
+- Updated the Docker image to: *TODO FILL IN ONCE MERGING THE NEW DOCKER IMAGE*.
```
Comment (ShirleyDenkberg) [inline Packs/CortexDataLake/ReleaseNotes/1_4_12.md:6]: Put this bullet last and don't forget to complete this.

**Comment by ShirleyDenkberg** [comment]
@omerKarkKatz Doc review completed.

File: Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py [hunk 3]
```diff
@@ -244,11 +250,19 @@ def query_loggings(self, query: str) -> tuple[list[dict], list]:
                 error_message = query_result
 
             raise DemistoException(f'Error in query to Cortex Data Lake XSOAR Connector [{status_code}] - {error_message}')
-
+        raw_results = []
         try:
-            raw_results = [r.json() for r in query_service.iter_job_results(job_id=query_result.get('jobId'),
-                                                                            result_format='valuesDictionary',
-                                                                            max_wait=2000)]
+            for r in query_service.iter_job_results(job_id=query_result.get(
```
Comment (omerKarkKatz) [inline Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py:255]: Do we have any issues with page_number/page_size entered as None to the iter_job_results?

File: Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake_test.py [hunk 4]
```diff
@@ -196,7 +196,7 @@ def test_query_logs_command_transform_results_1():
     cdl_records_xform = load_test_data('./test_data/test_query_logs_command_transform_results_xformed.json')
 
     class MockClient():
-        def query_loggings(self, query):
+        def query_loggings(self, query, page_number=None, page_size=None):
```
Comment (omerKarkKatz) [inline Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake_test.py:199]: What does this addition add to the test?
How does it test the new use cases you entered?

**Review by omerKarkKatz — COMMENTED** [review]
Nice work.
Please look at the review comments..

File: Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py (hunk 3 above)
Comment (mayyagoldman) [reply Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake.py:255]: image
Should be fine - the parameters are included internally only if they rnt None

File: Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake_test.py (hunk 4 above)
Comment (mayyagoldman) [reply Packs/CortexDataLake/Integrations/CortexDataLake/CortexDataLake_test.py:199]: The signature of the mock method had to be modified to match the original method

**Pull request merged**